{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5164b88f-58f9-4c98-b1dd-451a2fcd60eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "################################################ VARIANTES GERMINALES ######################################################\n",
    "################################################  HAPLOTYPECALLER ######################################################\n",
    "\n",
    "## LLAMADO DE VARIANTES\n",
    "reference=\"/home/yordonez3/Danny_OV_WES/Concatenated_fastq/GRCh38_Gencode31/hg38.fa\"\n",
    "output_dir=\"/home/yordonez3/Danny_OV_WES/Concatenated_fastq/EXOMA/variantes\"\n",
    "dbsnp_file=\"/home/yordonez3/Danny_OV_WES/GATK_recursos/Homo_sapiens_assembly38.dbsnp138.vcf\"\n",
    "num_threads=14\n",
    "\n",
    "mkdir -p \"$output_dir/gvcfs\"\n",
    "\n",
    "for bam_file in \"$output_dir\"/*-BQSR.bam; do\n",
    "  sample_name=$(basename \"$bam_file\" -BQSR.bam)\n",
    "\n",
    "  gatk --java-options \"-Xmx35g\" HaplotypeCaller \\\n",
    "    -R \"$reference\" \\\n",
    "    -I \"$bam_file\" \\\n",
    "    -O \"$output_dir/gvcfs/${sample_name}.g.vcf.gz\" \\\n",
    "    -ERC GVCF \\\n",
    "    --dbsnp \"$dbsnp_file\" \\\n",
    "    --native-pair-hmm-threads \"$num_threads\"\n",
    "    \n",
    "  echo \"HaplotypeCaller completado para $sample_name.\"\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7e2807cb-c45c-4e9d-b69e-87e04731323a",
   "metadata": {},
   "outputs": [],
   "source": [
    "## GENOMIC IMPORT\n",
    "\n",
    "reference=reference=\"/home/yordonez3/Danny_OV_WES/concatenated_fastq/concatenated_all_fastq/hg38.fa\"\n",
    "output_dir=\"reference=\"/home/yordonez3/Danny_OV_WES/concatenated_fastq/VARIANTES/gvcfs\"\n",
    "\n",
    "inputs=\"\"\n",
    "for file in $output_dir/gvcfs/*.g.vcf.gz; do\n",
    "  inputs+=\" -V $file\"\n",
    "done\n",
    "\n",
    "gatk --java-options \"-Xmx4g -Xms4g\" GenomicsDBImport \\\n",
    "  $inputs \\\n",
    "  --genomicsdb-workspace-path my_database \\\n",
    "  --tmp-dir $output_dir/gvcfs/ \\\n",
    "  --intervals chr1 \\\n",
    "  --intervals chr2 \\\n",
    "  --intervals chr3 \\\n",
    "  --intervals chr4 \\\n",
    "  --intervals chr5 \\\n",
    "  --intervals chr6 \\\n",
    "  --intervals chr7 \\\n",
    "  --intervals chr8 \\\n",
    "  --intervals chr9 \\\n",
    "  --intervals chr10 \\\n",
    "  --intervals chr11 \\\n",
    "  --intervals chr12 \\\n",
    "  --intervals chr13 \\\n",
    "  --intervals chr14 \\\n",
    "  --intervals chr15 \\\n",
    "  --intervals chr16 \\\n",
    "  --intervals chr17 \\\n",
    "  --intervals chr18 \\\n",
    "  --intervals chr19 \\\n",
    "  --intervals chr20 \\\n",
    "  --intervals chr21 \\\n",
    "  --intervals chr22 \\\n",
    "  --intervals chrX \\\n",
    "  --intervals chrY\n",
    "    \n",
    "echo \"GenomicsDBImport completado.\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ef6a771e-d6a6-4e36-b7eb-86a9eb339494",
   "metadata": {},
   "outputs": [],
   "source": [
    "## GENOTIPIFICACION\n",
    "\n",
    "reference=\"/home/yordonez3/Danny_OV_WES/concatenated_fastq/concatenated_all_fastq/hg38.fa\"\n",
    "gatk GenotypeGVCFs \\\n",
    "    -R \"$reference\" \\\n",
    "    -V gendb://my_database \\\n",
    "    -O genotipos_exo_27.vcf\n",
    "\n",
    "# Ver el archivo VCF y contar variantes\n",
    "bcftools view genotipos_chr1.vcf.gz | head -n 20\n",
    "bcftools view -H -v snps genotipos_chr1.vcf.gz | wc -l\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "27069667-19d8-4b35-b53a-c2fd8acd405a",
   "metadata": {},
   "outputs": [],
   "source": [
    "## 10. FILTRADO DE VARIANTES\n",
    "\n",
    "reference=\"/home/yordonez3/Danny_OV_WES/concatenated_fastq/concatenated_all_fastq/hg38.fa\"\n",
    "\n",
    "gatk VariantFiltration \\\n",
    "     -R \"$reference\" \\\n",
    "     -V genotipos_exo_27.vcf  \\\n",
    "     --window 35 \\\n",
    "     --cluster 3 \\\n",
    "     --filter-name \"FS\" \\\n",
    "     --filter \"FS > 30.0\" \\\n",
    "     --filter-name \"QD\" \\\n",
    "     --filter \"QD < 2.0\" \\\n",
    "     --filter-name \"MQ\" \\\n",
    "     --filter \"MQ < 30.0\" \\\n",
    "     -O geno_exo_57_filtered.vcf.g\n",
    "\n",
    "\n",
    "## cerrando el pc\n",
    "#### reference=\"/home/wovalle3/Samples/GRCh38_Gencode31/hg38.fa\"\n",
    "\n",
    "reference=\"/home/yordonez3/Danny_OV_WES/concatenated_fastq/concatenated_all_fastq/hg38.fa\"\n",
    "\n",
    "#### nohup gatk VariantFiltration \\\n",
    "\n",
    "gatk VariantFiltration \\\n",
    "     -R \"$reference\" \\\n",
    "     -V genotipos_exo_27.vcf  \\\n",
    "     --window 35 \\\n",
    "     --cluster 3 \\\n",
    "     --filter-name \"FS\" \\\n",
    "     --filter \"FS > 30.0\" \\\n",
    "     --filter-name \"QD\" \\\n",
    "     --filter \"QD < 2.0\" \\\n",
    "     --filter-name \"MQ\" \\\n",
    "     --filter \"MQ < 30.0\" \\\n",
    "     -O geno_exo_27_filtered.vcf.g\n",
    "\n",
    "   ####  > variantfiltration.log 2>&1 &\n",
    "\n",
    "# Contar los SNPs obtenidos\n",
    "bcftools view -H -v snps geno_exo_27_filtered.vcf.gz | wc -l\n",
    "# 8.999.354\n",
    "# 26.335.989\n",
    "# 30.338.751\n",
    "\n",
    "#Criterios para el filtrado\n",
    "FS (Phred-scaled p-value using Fisher's exact test to detect strand bias)\n",
    "FS > 30.0: Filters variants with a Fisher Strand (FS) score greater than 30.0.\n",
    "\n",
    "QD es el valor de calidad de una variante dividido por la profundidad de cobertura en ese sitio específico\n",
    "QD < 2.0: Filters variants with a Quality by Depth (QD) score less than 2.0.\n",
    "variantes con un valor QD menor que 2.0 serán marcadas como filtradas.\n",
    "    \n",
    "MQ Mapping Quality MQ ≥ 30, es un umbral comúnmente utilizado en muchos estudios. \n",
    "Se considera que un MQ de 30 o más indica una alineación confiable."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c6e864ea-7b9c-455b-a186-e6a3261974cc",
   "metadata": {},
   "outputs": [],
   "source": [
    "###PLINK\n",
    "\n",
    "mv geno_exo_27_filtered.vcf.gz geno_exo_27_filtered.vcf.gz.tbi /home/yordonez3/Danny_OV_WES/concatenated_fastq/EXOMA/variantes/plink\n",
    "\n",
    "#Antes de empezar revisar que el archivo tenga los IDs de los SNPs\n",
    "bcftools view --no-header geno_exo_27_filtered.vcf.gz | head\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6f8463e4-386f-4df3-8cb3-8ab2d00fa5c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "###PLINK\n",
    "VCF de entrada: geno_exo_57_filtered.vcf.gz\n",
    "\n",
    "bcftools view --no-header geno_exo_57_filtered.vcf.gz | head\n",
    "\n",
    "##1. Generar los bfiles con el archivo .vcf de mis muestras\n",
    "The first step for quality control (QC) analysis in PLINK starting from\n",
    "your VCF files is to convert the VCF file to PLINK format (BED, BIM, FAM).\n",
    "PLINK provides a tool called plink2 for this purpose. \n",
    "                                 \n",
    "plink --vcf geno_exo_57_filtered.vcf.gz --make-bed --out geno_plink\n",
    "\n",
    "#Results 28 samples\n",
    "Total genotyping rate is 0.461463.\n",
    "14.853.128 variants and 18 people pass filters and QC.\n",
    "\n",
    "Total genotyping rate is 0.434183.\n",
    "20.581.021 variants and 28 people pass filters and QC.\n",
    "\n",
    "\n",
    "#results 46 samples\n",
    "Total genotyping rate is 0.397116.\n",
    "28.508.607 variants and 46 people pass filters and QC.\n",
    "Note: No phenotypes present.\n",
    "    \n",
    "#results 57 samples\n",
    "Total genotyping rate is 0.38405.\n",
    "32.742.195 variants and 57 people pass filters and QC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c6ad023f-140e-4f76-a919-ab4bc11f8481",
   "metadata": {},
   "outputs": [],
   "source": [
    "#2. Perform QC analysis\n",
    "\n",
    "#Opciones de QC\n",
    "#Removing individuals with high missing genotype rates (--mind): --mind 0.05\n",
    "#Removing SNPs with high missingness (--geno): --geno 0.05\n",
    "#Removing SNPs with low minor allele frequency (--maf): --maf 0.01\n",
    "#Performing Hardy-Weinberg equilibrium (HWE) testing (--hwe): 1e-6\n",
    "#Removing SNPs with extreme heterozygosity (--het).\n",
    "#Identificación de SNPs Duplicados (--list-duplicate-vars)\n",
    "#Filtrar por Identidad por Descendencia (IBD) (--genome y --rel-cutoff)\n",
    "\n",
    "#Usar comando geno 0.1 para que deje solamente SNPs con CR>90%\n",
    "plink --bfile geno_plink --geno 0.1  --make-bed --out geno_plink_fil1\n",
    "\n",
    "#Results\n",
    "Total genotyping rate is 0.461463.\n",
    "12.763.929 variants removed due to missing genotype data (--geno).\n",
    "2.089.199 variants and 18 people pass filters and QC.\n",
    "\n",
    "Total genotyping rate is 0.434183.\n",
    "18.721.953 variants removed due to missing genotype data (--geno).\n",
    "1.859.068 variants and 28 people pass filters and QC.\n",
    "\n",
    "#results 46\n",
    "Total genotyping rate is 0.397116.\n",
    "25.802.351 variants removed due to missing genotype data (--geno).\n",
    "2.706.256 variants and 46 people pass filters and QC.\n",
    "Note: No phenotypes present.\n",
    "    \n",
    "#57\n",
    "Total genotyping rate is 0.38405.\n",
    "29.499.078 variants removed due to missing genotype data (--geno).\n",
    "3.243.117 variants and 57 people pass filters and QC.\n",
    "\n",
    "#Control de calidad para individuos con mind. Deja solamente casos CR>90%\n",
    "plink --bfile geno_plink_fil1 --mind 0.25 --make-bed --out geno_plink_fil2   ##Mejor hacerlo con 0.1 despues\n",
    "\n",
    "#Results\n",
    "Total genotyping rate in remaining samples is 0.943978.\n",
    "2.089.199 variants and 17 people pass filters and QC.\n",
    "\n",
    "Total genotyping rate is 0.97497.\n",
    "1.859.068 variants and 28 people pass filters and QC.\n",
    "\n",
    "#results 46\n",
    "Total genotyping rate is 0.972913.\n",
    "2.706.256 variants and 46 people pass filters and QC.\n",
    "\n",
    "#57\n",
    "Total genotyping rate is 0.973838.\n",
    "3243117 variants and 57 people pass filters and QC."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a3e85e89-420e-42fc-a449-ac084425bab4",
   "metadata": {},
   "outputs": [],
   "source": [
    "#3. Filtrar por MAF y HW\n",
    "\n",
    "#MAF: 0.01 remueve variantes con frecuencia de alelo menor a 1% (es dcir mutaciones y demas)\n",
    "plink --bfile geno_plink_fil2 --maf 0.01 --make-bed --out geno_plink_fil3\n",
    "\n",
    "#Results\n",
    "Total genotyping rate is 0.943978\n",
    "124.280 variants removed due to minor allele threshold(s)\n",
    "1.964.919 variants and 17 people pass filters and QC.\n",
    "\n",
    "Total genotyping rate is 0.97497.\n",
    "3041 variants removed due to minor allele threshold(s)\n",
    "1.856.027 variants and 28 people pass filters and QC\n",
    "\n",
    "#results 46\n",
    "Total genotyping rate is 0.972913.\n",
    "2042 variants removed due to minor allele threshold(s)\n",
    "(--maf/--max-maf/--mac/--max-mac).\n",
    "2.704.214 variants and 46 people pass filters and QC.\n",
    "\n",
    "#57\n",
    "Total genotyping rate is 0.973838.\n",
    "2188097 variants removed due to minor allele threshold(s)\n",
    "(--maf/--max-maf/--mac/--max-mac).\n",
    "1.055.020 variants and 57 people pass filters and QC.\n",
    "\n",
    "\n",
    "#HW\n",
    "plink --bfile geno_plink_fil3 --hwe 1e-6 --hwe-all --make-bed --out geno_plink_fil4\n",
    "\n",
    "#Results\n",
    "Total genotyping rate is 0.944006.\n",
    "--hwe: 960 variants removed due to Hardy-Weinberg exact test.\n",
    "1.963.959 variants and 17 people pass filters and QC.\n",
    "\n",
    "Total genotyping rate is 0.974968.\n",
    "--hwe: 520 variants removed due to Hardy-Weinberg exact test.\n",
    "1.855.507 variants and 28 people pass filters and QC.\n",
    "\n",
    "#results 46\n",
    "Total genotyping rate is 0.972912.\n",
    "--hwe: 1333 variants removed due to Hardy-Weinberg exact test.\n",
    "2.702.881 variants and 46 people pass filters and QC.\n",
    "\n",
    "#57\n",
    "Total genotyping rate is 0.968644.\n",
    "--hwe: 2072 variants removed due to Hardy-Weinberg exact test.\n",
    "1.052.948 variants and 57 people pass filters and QC."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ae835dc0-f9b5-4c62-8b87-eca28a75caad",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Perform LD pruning in two steps:\n",
    "# Step 1: Generate list of SNPs to keep\n",
    "# We perform linkage disequilibrium (LD) pruning using:\n",
    "  # - a window size of 50 SNPs\n",
    "  # - a step size of 5 SNPs\n",
    "  # - r2 threshold of 0.2 (correlation between SNPs)\n",
    "\n",
    "plink --bfile geno_plink_fil4 --indep-pairwise 50 5 0.2 --out ld_pruned\n",
    "\n",
    "#Results \n",
    "Total genotyping rate is 0.944005.\n",
    "1.963.959 variants and 17 people pass filters and QC.\n",
    "Pruning complete.  1.668.064 of 1.963.959 variants removed.\n",
    "\n",
    "#results 46\n",
    "Pruning complete.  1848160 of 2702881 variants removed.\n",
    "\n",
    "\n",
    "# Step 2: Apply LD pruning\n",
    "plink --bfile geno_plink_fil4 --extract ld_pruned.prune.in --make-bed --out geno_plink_fil5\n",
    "\n",
    "#Results \n",
    "Ambiguous sex IDs written to geno_plink_fil5.nosex .\n",
    "--extract: 1963959 variants remaining.\n",
    "Warning: At least 295894 duplicate IDs in --extract file.\n",
    "Total genotyping rate is 0.944005.\n",
    "1.963.959 variants and 17 people pass filters and QC.\n",
    "\n",
    "Total genotyping rate is 0.974963.\n",
    "1.855.507 variants and 28 people pass filters and QC\n",
    "\n",
    "#results 46\n",
    "Total genotyping rate is 0.972908.\n",
    "2.702.881 variants and 46 people pass filters and QC.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4488fa2e-3a0a-4da7-8519-bbc15461774f",
   "metadata": {},
   "outputs": [],
   "source": [
    "#4. Set-missing-var-ids para mis datos y el panel de referencia\n",
    "\n",
    "#El comando set-missing-var-ids en PLINK se utiliza para asignar \n",
    "#identificadores (IDs) a los SNPs que no tienen un ID en el archivo BIM\n",
    "\n",
    "plink --bfile geno_plink_fil5 --set-missing-var-ids chr@:#:\\$2:\\$1 --make-bed --out genoexo_NoMissVar\n",
    "    \n",
    "#Results\n",
    "Total genotyping rate is 0.944005.\n",
    "1.963.959 variants and 17 people pass filters and QC.\n",
    "\n",
    "Total genotyping rate is 0.974963.\n",
    "1.855.507 variants and 28 people pass filters and QC.\n",
    "\n",
    "#results 46\n",
    "Total genotyping rate is 0.972908.\n",
    "2.702.881 variants and 46 people pass filters and QC.\n",
    "\n",
    "#57\n",
    "Total genotyping rate is 0.968623.\n",
    "1.052.948 variants and 57 people pass filters and QC.\n",
    "\n",
    "\n",
    "#Esto lo hago desde la carpeta de FinalReferencePanel si no se tiene \n",
    "#plink --bfile extractedChrAll --set-missing-var-ids chr@:#:\\$2:\\$1 --make-bed --out extractedChrAll_NoMissVar\n",
    "    \n",
    "#Results\n",
    "23.886.383 variants loaded from .bim file.\n",
    "969 people (0 males, 0 females, 969 ambiguous) loaded from .fam.\n",
    "Total genotyping rate is 0.999872.\n",
    "23.886.383 variants and 969 people pass filters and QC.\n",
    "\n",
    "#Mover los archivos a la carpeta desde donde se esta haciendo el analisis\n",
    "mv extractedChrAll_NoMissVar* /home/wovalle3/Samples/total_samples_pool/EXOMA/variantes/plink"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b17c5184-7101-497b-86fd-47947d861c4a",
   "metadata": {},
   "outputs": [],
   "source": [
    "#5. Encontrar variantes que se comparten entre la base de referencia y la mia\n",
    "\n",
    "#Para asegurarte de que el panel de referencia tiene exactamente las mismas \n",
    "#variantes que tu archivo antes de realizar un merge en PLINKls\n",
    "\n",
    "comm -12 <(sort genoexo_NoMissVar.bim) <(sort extractedChrAll_NoMissVar.bim) > variantes_comunes.txt\n",
    "wc -l variantes_comunes.txt #119.776 compartidas  #111.293   #116.467 #97.632\n",
    "\n",
    "plink --bfile extractedChrAll_NoMissVar --extract variantes_comunes.txt --make-bed --out Ref_comunes\n",
    "\n",
    "#Results\n",
    "Total genotyping rate is 0.99987.\n",
    "119.776 variants and 969 people pass filters and QC\n",
    "\n",
    "Total genotyping rate is 0.999874.\n",
    "111.293 variants and 969 people pass filters and QC.\n",
    "\n",
    "#results 46\n",
    "Total genotyping rate is 0.999877.\n",
    "116.467 variants and 969 people pass filters and QC.\n",
    "\n",
    "#57\n",
    "Total genotyping rate is 0.999876.\n",
    "97.632 variants and 969 people pass filters and QC.\n",
    "\n",
    "\n",
    "\n",
    "plink --bfile genoexo_NoMissVar --extract variantes_comunes.txt --make-bed --out genoexo_comunes\n",
    "#Results\n",
    "Total genotyping rate is 0.95625.\n",
    "119.776 variants and 17 people pass filters and QC.\n",
    "\n",
    "Total genotyping rate is 0.978565.\n",
    "111.293 variants and 28 people pass filters and QC.\n",
    "\n",
    "#results 46\n",
    "Total genotyping rate is 0.975627.\n",
    "116.467 variants and 46 people pass filters and QC.\n",
    "\n",
    "#57\n",
    "Total genotyping rate is 0.975575.\n",
    "97.632 variants and 57 people pass filters and QC."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e5c884ff-9e60-4d3a-b429-5adef01b1699",
   "metadata": {},
   "outputs": [],
   "source": [
    "#6. Merge del panel de referencia con mis datos\n",
    "\n",
    "plink --bfile Ref_comunes --bmerge genoexo_comunes --make-bed --out merged_set_28\n",
    "\n",
    "(969 + 17)\n",
    "Total genotyping rate is 0.999118.\n",
    "119.776 variants and 986 people pass filters and QC.\n",
    "\n",
    "Total genotyping rate is 0.999275.\n",
    "111.293 variants and 997 people pass filters and QC.\n",
    "\n",
    "#results 46\n",
    "Total genotyping rate is 0.998778.\n",
    "116.467 variants and 1015 people pass filters and QC.\n",
    "\n",
    "#57\n",
    "Total genotyping rate is 0.998526.\n",
    "97.632 variants and 1026 people pass filters and QC.\n",
    "\n",
    "\n",
    "plink --bfile merged_set_28 --geno 0.10  --make-bed --out merged_set_fil28\n",
    "\n",
    "#results\n",
    "0 variants removed due to missing genotype data (--geno).\n",
    "119776 variants and 986 people pass filters and QC.\n",
    "\n",
    "0 variants removed due to missing genotype data (--geno).\n",
    "111293 variants and 997 people pass filters and QC\n",
    "\n",
    "0 variants removed due to missing genotype data (--geno).\n",
    "116467 variants and 1015 people pass filters and QC.\n",
    "\n",
    "0 variants removed due to missing genotype data (--geno).\n",
    "97632 variants and 1026 people pass filters and QC\n",
    "\n",
    "\n",
    "#PCA con los resultados\n",
    "plink --bfile merged_set_fil28 --pca --out pca_results_57\n",
    "#mac\n",
    "scp -r wovalle3@bio-compjord01.biosci.gatech.edu:/home/wovalle3/Samples/total_samples_pool/EXOMA/variantes/plink/pca_results_57.eigenval /Users/wen/Documents/\n",
    "scp -r wovalle3@bio-compjord01.biosci.gatech.edu:/home/wovalle3/Samples/total_samples_pool/EXOMA/variantes/plink/pca_results_57.eigenvec /Users/wen/Documents/\n",
    "\n",
    "#pc INC\n",
    "scp -r wovalle3@bio-compjord01.biosci.gatech.edu:/home/wovalle3/Samples/total_samples_pool/EXOMA/variantes/plink/pca_results_28.eigenval C:\\Users\\wmontero\\Documents\n",
    "scp -r wovalle3@bio-compjord01.biosci.gatech.edu:/home/wovalle3/Samples/total_samples_pool/EXOMA/variantes/plink/pca_results_28.eigenvec C:\\Users\\wmontero\\Documents\n",
    "#Los archivos generados por PLINK para el análisis de PCA (pca_results.eigenvec y pca_results.eigenval) \n",
    "#son archivos de texto que contienen los resultados del análisis de componentes principales. \n",
    "#Puedes abrir y analizar estos archivos en R para visualización y análisis adicionales"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "65a11625-746c-42ee-be4d-1369e339eb91",
   "metadata": {},
   "outputs": [],
   "source": [
    "#9.\tGenerar un nuevo archivo recodificado para Admixture\n",
    "\n",
    "plink --bfile merged_set_fil28 --recode12 --out Final_admixture57\n",
    "\n",
    "#results\n",
    "Total genotyping rate is 0.999118.\n",
    "119.776 variants and 986 people pass filters and QC.\n",
    "\n",
    "Total genotyping rate is 0.999275.\n",
    "111.293 variants and 997 people pass filters and QC.\n",
    "\n",
    "#results 46\n",
    "Total genotyping rate is 0.998778.\n",
    "116.467 variants and 1015 people pass filters and QC.\n",
    "\n",
    "#57\n",
    "Total genotyping rate is 0.998526.\n",
    "97.632 variants and 1026 people pass filters and QC.\n",
    "\n",
    "#Instalar Admixture en el servidor \n",
    "wget https://dalexander.github.io/admixture/binaries/admixture_linux-1.3.0.tar.gz\n",
    "\n",
    "#Descomprimir el archivo tar.gz\n",
    "tar -zxvf admixture_linux-1.3.0.tar.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8e713253-2b89-4588-acd2-d1a734d6d0a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "#10. Correr admixture desde donde esta el archivo ped\n",
    "/home/wovalle3/ADMIXTURE/dist/admixture_linux-1.3.0/admixture Final_admixture57.ped 3\n",
    "\n",
    "#Results\n",
    "Summary: \n",
    "Converged in 13 iterations (239.995 sec)\n",
    "Loglikelihood: -85308375.284505\n",
    "Fst divergences between estimated populations: \n",
    "\tPop0\tPop1\t\n",
    "Pop0\t\n",
    "Pop1\t0.136\t\n",
    "Pop2\t0.181\t0.123\t\n",
    "Writing output files.\n",
    "\n",
    "Converged in 16 iterations (249.68 sec)\n",
    "Loglikelihood: -71583396.008181\n",
    "Fst divergences between estimated populations: \n",
    "\tPop0\tPop1\t\n",
    "Pop0\t\n",
    "Pop1\t0.134\t\n",
    "Pop2\t0.182\t0.126\t\n",
    "Writing output files.\n",
    "\n",
    "#results 46\n",
    "\n",
    "\n",
    "#Pasar archivo Final_admixture.3.Q y merged_set.fam a pc para mirar resultados\n",
    "#mac\n",
    "scp -r wovalle3@bio-compjord01.biosci.gatech.edu:/home/wovalle3/Samples/total_samples_pool/EXOMA/variantes/plink/Final_admixture57.3.Q /Users/wen/Documents/\n",
    "scp -r wovalle3@bio-compjord01.biosci.gatech.edu:/home/wovalle3/Samples/total_samples_pool/EXOMA/variantes/plink/merged_set_28.fam /Users/wen/Documents/\n",
    "\n",
    "#pc INC\n",
    "scp -r wovalle3@bio-compjord01.biosci.gatech.edu:/home/wovalle3/Samples/total_samples_pool/EXOMA/variantes/plink/Final_admixture28.3.Q C:\\Users\\wmontero\\Documents\n",
    "scp -r wovalle3@bio-compjord01.biosci.gatech.edu:/home/wovalle3/Samples/total_samples_pool/EXOMA/variantes/plink/merged_set_28.fam C:\\Users\\wmontero\\Documents"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
